multigeiger/multigeiger.ino

```diff
@@ -106,138 +103,146 @@ int update_ble_status(void) {  // currently no error detection
   return st;
 }
 
-void publish(unsigned long current_ms, unsigned long current_counts, unsigned long gm_count_timestamp, unsigned long current_hv_pulses,
-             float temperature, float humidity, float pressure) {
-  static unsigned long last_timestamp = millis();
-  static unsigned long last_counts = 0;
-  static unsigned long last_hv_pulses = 0;
-  static unsigned long last_count_timestamp = 0;
+void read_THP(unsigned long current_ms,
+              bool *have_thp, float *temperature, float *humidity, float *pressure) {
+  static unsigned long last_timestamp = 0;
+  // first call: immediately query thp sensor
+  // subsequent calls: only query every HEARTBEAT_INTERVAL
+  if (!last_timestamp || ((current_ms - last_timestamp) >= (HEARTBEAT_INTERVAL * 1000))) {
+    *have_thp = read_thp_sensor(temperature, humidity, pressure);
+    last_timestamp = current_ms;
+  }
+}
+
+void local_alarmsequence(void) {
+  // call all alarm action options
+  // currently local beep every display refresh
+  alarm();
+}
+
+void process_GMC(unsigned long current_ms, unsigned long current_counts, unsigned long gm_count_timestamp, unsigned long current_hv_pulses,
+                 bool have_thp, float temperature, float humidity, float pressure, int wifi_status) {
+  struct GM_State {
+    unsigned long timestamp;
+    unsigned long counts;
+    unsigned long last_count_timestamp;
+    unsigned long hv_pulses;
+  };
+
+  // Events
+  const byte HEARTBEAT = 0;
+  const byte MEASUREMENT = 1;
+  const byte ONE_MINUTE = 2;
+
+  const int savedstates_count = 3;
+
+  static GM_State saved_state[savedstates_count];
```

Comment:
are the array members in a defined state now, like e.g. is it documented that this is all-zero?